acquia/mc-cs-plugin-custom-objects
# MAUT-4460: New Feature : Ability to add ‘ Exclusion lists’ at time of segment build

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | / 
| Issues addressed (#s or URLs) | https://backlog.acquia.com/browse/MAUT-4460
| BC breaks? | N
| Deprecations? | N 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:


[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. 
2. 

#### Steps to test this PR:
1) Load up [MAUT-4460](https://backlog.acquia.com/browse/MAUT-4460)
2) create 2 contacts with "testcontact1@acquia.com" and "testcontact2@gmail.com" in email
3) create custom object "Email List" with "testcontact1@acquia.com" and "testcontact2@mautic.com" as items(_you have to enable custom objects functionality if not already enabled_)
4) In segment filters you should be able to see "not in custom objects" in operator list
5) create a segment with filter : email  > not in custom objects > select above custom Object 
6) Once the segment builds in background, you should be able to see contact with "testcontact2@gmail.com" as email added inside the segment
7) Contact with "testcontact1@acquia.com" as email was not added as it was in 'exclude list' i.e., it was present in custom object

#### Other areas of Mautic that may be affected by the change:
1. 
2. 

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
